### PR TITLE
CLN: remove unused legacy pickle compat code

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/core/generic.py [hunk 1]
```diff
@@ -1928,9 +1928,9 @@ def __setstate__(self, state):
                         object.__setattr__(self, k, v)
 
             else:
-                self._unpickle_series_compat(state)
+                raise NotImplementedError("Pre-0.12 pickles are no longer supported")
```
Comment (WillAyd) [inline pandas/core/generic.py]: Actually might want to change the message to `Pre-0.20.3` pickles consistently

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by jreback** [comment]
hit wrong milestone

**Pull request merged**